src/providers/azure-ai-inference/api.ts

```diff
@@ -0,0 +1,78 @@
+import { GITHUB } from '../../globals';
+import { ProviderAPIConfig } from '../types';
+
+const AzureAIInferenceAPI: ProviderAPIConfig = {
+  getBaseURL: ({ providerOptions }) => {
+    const {
+      azureDeploymentName,
+      azureRegion,
+      azureDeploymentType,
+      provider,
+      azureEndpointName,
+    } = providerOptions;
+    if (provider === GITHUB) {
+      return 'https://models.inference.ai.azure.com';
+    }
+    if (azureDeploymentType === 'serverless') {
+      return `https://${azureDeploymentName?.toLowerCase()}.${azureRegion}.models.ai.azure.com`;
+    }
+
+    return `https://${azureEndpointName}.${azureRegion}.inference.ml.azure.com/score`;
+  },
+  headers: ({ providerOptions }) => {
+    const { apiKey, azureDeploymentType, azureDeploymentName } =
+      providerOptions;
+    const headers: Record<string, string> = {
+      Authorization: `Bearer ${apiKey}`,
+      'extra-parameters': 'ignore',
```

Comment:
@b4s36t4 This seem to make the azure-ai provider fail at the moment:
```
Error code: 400 - {'detail': "Unexpected EXTRA_PARAMETERS option ignore, expected options are ['error', 'drop', 'pass-through']", 'provider': 'azure-ai'}
```
I tried cloning the project and replacing it with `drop` to see if it fixes it it but the provider isn't available when running the project locally so I'm not sure how to proceed here.

Reply:
@Tug is it possible to share the request as cURL for me to test in my local. Would need to replicate the issue from my side as well.

Reply:
Hey, raised a PR https://github.com/Portkey-AI/gateway/pull/901. This should ideally give you the capability of passing the `extra-params` from the headers with request, with default as `pass-through`

You can use `x-portkey-azure-extra-params` header to specify the value for `extra-parameters`, I'll share an update with you once it's merged and released.

Reply:
> @Tug is it possible to share the request as cURL for me to test in my local. Would need to replicate the issue from my side as well.

Sure it's as simple as
```
curl -X POST "https://api.portkey.ai/v1/chat/completions" \
-H "Content-Type: application/json" \
-H "Authorization: Bearer REDACTED" \
-H "x-portkey-config: REDACTED" \
-H "x-portkey-provider: azure-ai" \
-H "x-portkey-virtual-key: REDACTED" \
-d '{
  "messages": [
    {
      "role": "system",
      "content": "You are a helpful assistant."
    },
    {
      "role": "user",
      "content": "How many languages are in the world?"
    }
  ],
  "temperature": 0,
  "stream": false
}'
```
The virtual key is configured for an azure AI serverless deployment.

Reply:
Thanks for the response, as I have stated above the PR needs approval, will update you as soon as it landed in the release.